code/game/machinery/vending.dm

```diff
@@ -607,9 +607,15 @@
 	if (!message)
 		return
 
-	for(var/mob/O in hearers(src, null))
-		O.show_message("<span class='game say'><span class='name'>\The [src]</span> beeps, \"[message]\"</span>",2)
-	return
+	var/list/hear = hearers(src, null)
+	var/list/client/clients = list()
+	for (var/mob/M in hear)

```

Comment:
Can't this be refactored into `visible_message()` ? Or nag due to it spoofing normal speaking?